finality-aleph/src/testing/mocks/validator_network.rs

```diff
@@ -90,3 +112,373 @@ impl<D: Data> MockNetwork<D> {
+        Pin::new(&mut self.get_mut().stream).poll_shutdown(cx)
+    }
+}
+
+impl AsyncRead for UnreliableDuplexStream {
+    fn poll_read(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<IoResult<()>> {
+        Pin::new(&mut self.get_mut().stream).poll_read(cx, buf)
+    }
+}
+
+/// A stream that can be split into two instances of UnreliableDuplexStream.
+#[derive(Debug)]
+pub struct UnreliableSplittable {
+    incoming_data: UnreliableDuplexStream,
+    outgoing_data: UnreliableDuplexStream,
+}
+
+impl UnreliableSplittable {
+    /// Create a pair of mock splittables connected to each other.
+    pub fn new(max_buf_size: usize, ends_after: usize) -> (Self, Self) {
+        let (in_a, out_b) = duplex(max_buf_size);
+        let (in_b, out_a) = duplex(max_buf_size);
+        (
+            UnreliableSplittable {
+                incoming_data: UnreliableDuplexStream {
+                    stream: in_a,
+                    counter: ends_after,
+                },
+                outgoing_data: UnreliableDuplexStream {
+                    stream: out_a,
+                    counter: ends_after,
+                },
+            },
+            UnreliableSplittable {
+                incoming_data: UnreliableDuplexStream {
+                    stream: in_b,
+                    counter: ends_after,
+                },
+                outgoing_data: UnreliableDuplexStream {
+                    stream: out_b,
+                    counter: ends_after,
+                },
+            },
+        )
+    }
+}
+
+impl AsyncRead for UnreliableSplittable {
+    fn poll_read(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<IoResult<()>> {
+        Pin::new(&mut self.get_mut().incoming_data).poll_read(cx, buf)
+    }
+}
+
+impl AsyncWrite for UnreliableSplittable {
+    fn poll_write(self: Pin<&mut Self>, cx: &mut Context<'_>, buf: &[u8]) -> Poll<IoResult<usize>> {
+        Pin::new(&mut self.get_mut().outgoing_data).poll_write(cx, buf)
+    }
+
+    fn poll_flush(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<IoResult<()>> {
+        Pin::new(&mut self.get_mut().outgoing_data).poll_flush(cx)
+    }
+
+    fn poll_shutdown(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<IoResult<()>> {
+        Pin::new(&mut self.get_mut().outgoing_data).poll_shutdown(cx)
+    }
+}
+
+impl Splittable for UnreliableSplittable {
+    type Sender = UnreliableDuplexStream;
+    type Receiver = UnreliableDuplexStream;
+
+    fn split(self) -> (Self::Sender, Self::Receiver) {
+        (self.outgoing_data, self.incoming_data)
+    }
+}
+
+type Address = u32;
+type Addresses = HashMap<AuthorityId, Vec<Address>>;
+type Callers = HashMap<AuthorityId, (MockDialer, MockListener)>;
+type Connection = UnreliableSplittable;
+
+#[derive(Clone)]
+pub struct MockDialer {
+    c_connect: mpsc::UnboundedSender<(Address, oneshot::Sender<Connection>)>,
```

Comment:
removing, but also changing `c_connect` to `channel_connect` and `c_new_connection` to `channel_accept` in `MockDialer` and `MockListener` respectively, since they already have similarly named methods